### PR TITLE
Accoil Analytics: Refactor postToAccoil into sub actions

### PR DESCRIPTION
**Pull request by accoilmj**
This updates our existing single action into separate action handlers for identify, group, track, page, screen, and adds unit tests for each. I'm hoping this is a simple in place upgrade as it is still private and not publicly available yet but let me know if not.

## Testing

Tested using the local harness.

- [x] Added [unit tests](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing) for new functionality
- [x] Tested end-to-end using the [local server](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing)
- [ ] [Segmenters] Tested in the staging environment

### CODE REVIEW

**Comment by joe-ayoub-segment** [comment]
Thanks @accoilmj I'll schedule for review

**Comment by accoilmj** [comment]
I see some test are failing. Looking into it.

**Comment by accoilmj** [comment]
Looks like a timestamp / zone issue in snapshots where they were drifting in my machine vs remote. I've pushed a fix.

**Comment by accoilmj** [comment]
I've pushed new changes that may fix the timezone drift problem.

**Comment by accoilmj** [comment]
It looks like all of the Accoil tests passed this time but now there's:

```
   Failed tasks:
   
   - @segment/actions-cli:test
```

**Comment by accoilmj** [comment]
The tests pass locally and looked to maybe be intermittent failures so I've rebased and pushed again to be sure.

File: packages/destination-actions/src/destinations/accoil-analytics/group/index.ts [hunk 1]
```diff
@@ -0,0 +1,80 @@
+import type { ActionDefinition } from '@segment/actions-core'
+import type { Settings } from '../generated-types'
+import type { Payload } from './generated-types'
+import { commonFields } from '../common-fields'
+import { endpointUrl } from '../utils'
+
+const action: ActionDefinition<Settings, Payload> = {
+  title: 'Group',
+  description: 'Identify Accounts (groups) in Accoil',
+  defaultSubscription: 'type = "identify"',
```
Comment (joe-ayoub-segment) [inline packages/destination-actions/src/destinations/accoil-analytics/group/index.ts:10]: should this be group?
```suggestion
  defaultSubscription: 'type = "group"',
```

**Review by joe-ayoub-segment — CHANGES_REQUESTED** [review]
looks good. Just a minor comment to address.

**Comment by accoilmj** [comment]
You're right. Good catch. I've pushed a fix for that.

**Comment by joe-ayoub-segment** [comment]
hi @accoilmj PR deployed

**Comment by accoilmj** [comment]
Thanks @joe-ayoub-segment

**Pull request merged**